### PR TITLE
Rosetta: Add models and network endpoints

### PR DESCRIPTION
**Pull request by Schmavery**
The modules in the `models` folder were generated using the Rosetta OpenAPI spec.
This PR contains a mostly proof-of-concept implementation of the network API. There are still several issues with it, but an attempt was made to get all the pieces hooked up and working together.

I still haven't totally finished up the dockerfile from https://github.com/CodaProtocol/coda/pull/5109, so I can't test it in that environment. The server itself seems to run ok locally, harder to get it tested with a working network + archive :P

TODO:
- ~~Convert to using Logger rather than print statements~~ done
- Try to get it building in docker?
- Fix/improve network endpoint logic
- Figure out the best place to get the genesis block info

### CODE REVIEW

File: src/app/rosetta/network.ml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            Block_identifier.create
+              ((latest_block#protocolState)#consensusState)#blockHeight
+              latest_block#stateHash
+        ; current_block_timestamp=
+            ((latest_block#protocolState)#blockchainState)#utcDate
+        ; genesis_block_identifier=
+            Block_identifier.create Int64.zero genesis_block_state_hash
+        ; peers=
+            (res#daemonStatus)#peers |> Array.to_list
+            |> List.map ~f:Peer.create }
```
Comment (bkase) [inline src/app/rosetta/network.ml:76]: graphql is legit

File: src/app/rosetta/network.ml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+let router ~graphql_uri (module Db : Caqti_async.CONNECTION) route body =
+  let open Async.Deferred.Result.Let_syntax in
+  match route with
+  | ["list"] ->
+      let%map _meta = Mappers.parse @@ Metadata_request.of_yojson body in
+      Network_list_response.to_yojson
+        { Network_list_response.network_identifiers=
+            [ { Network_identifier.blockchain= "coda"
+              ; network= "testnet"
+              ; sub_network_identifier= None } ] }
```
Comment (bkase) [inline src/app/rosetta/network.ml:46]: I recall us wanting to use sub_network_identifier for something, but I don't remember now. Do you know? Anyway, we shouldn't block the PR on that anyway.

File: src/app/rosetta/network.ml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  | ["status"] ->
+      (* TODO: Check that the network corresponds to the node we're connected to *)
+      let%bind _network = Mappers.parse @@ Network_request.of_yojson body in
+      let%bind res = Graphql.query (Get_status.make ()) graphql_uri in
+      let%bind latest_block =
+        Deferred.return
+          ( match res#bestChain with
```
Comment (bkase) [inline src/app/rosetta/network.ml:53]: should we be getting this block from the Db long term? Or do you think we should always cache the latest block via graphql or something.

Comment (Schmavery) [reply src/app/rosetta/network.ml:53]: yeah probably want to get it from the db

File: src/app/rosetta/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Implementation of the [Rosetta Node API](https://www.rosetta-api.org/docs/node_api_introduction.html) for Coda.
+
+To regenerate the models:
+```
+git clone https://github.com/coinbase/rosetta-specifications.git
+cd rosetta-specifications
+`brew install openapi`
```
Comment (psteckler) [inline src/app/rosetta/README.md:7]: is this Mac-specific?

Comment (Schmavery) [reply src/app/rosetta/README.md:7]: The instructions are but you should be able to install the openapi tool wherever. I’ll try to scare up some Linux install instructions

**Comment by bkase** [comment]
Landing this so we can start from somewhere

**Pull request merged**